cEP-0019.md

```diff
@@ -0,0 +1,454 @@
+>>> print(result_adjust)
+[0.23043478260869565, 0.1, 1.1]
+>>> result_rounded = [round(i, 3) for i in result_adjust]
+>>> print(result_rounded)
+[0.23, 0.1, 1.1]
+```
+
+Therefore, the k factor for Alice, Bob and Charlie in the next iteration would
+be: 0.23, 0.1 and 1.1, respectively.
+
+If a person has zeros for all factors, then he is regarded as an inactive
+reviewer and would not be displayed. If a person has negative score, he will
+still be displayed, but the meta-reviews he gives to other people will not
+count.
+
+#### Meta-reviews in Need
+
+##### Definition
+
+When is a review needing meta-review? There are two perspectives:
+
+1. Meta-review by developers, i.e. all people within the organization. The
+   'developer' concept here is different from that used in coala. Instead,
+   anyone with write access (including maintainers) to the repo should be
+   regarded as developers. For developers, review comments before merge/close
+   need meta-reviews. Those review comments are sorted according to update
+   time. In particular, one will always see the review comment that is under
+   his PR first.
+
+2. Meta-review by maintainers. For maintainers, review comments after
+   merge/close also need meta-reviews. Those meta-reviews will reflect the
+   accuracy of the review comments.
+
+##### Display
+
+Review comments that need meta-review will be displayed on a separate column
+on gh-board. There are three approaches:
+
+1. The worst case is that the tracking stores in the json which
+   PR do not have meta-reviews, and gh-board shows only a list of those PRs
+   (and the user must then find the reviews in GitHub PR web app, and do
+   meta-review there).
+
+2. An improvement is to store the identifier of the PR comments which
+   do not have meta-reviews and gh-board links to the PR review comment
+   in GitHub PR web app.
+
+3. In the ultimate version, gh-board would show the review comments.
+   In the future, people could do meta-reviews directly on gh-board. This also
+   provides an approach to doing meta-reviews anonymously.
+
+#### Performance Requirements
+
+Currently,
+[fetch-issues.js](https://github.com/coala/gh-board/blob/46f1ad5ce83d5d72aa95ea00703b92c2fc38b2dc/script/fetch-issues.js)
+pre-fetches issues and pull requests to speed up gh-board performance and
+save users' API hits. Meta-reviews are also to be fetched during this procedure.
+
+#### Other Features
+
+The GSoC project provides a blueprint for future enhancement. Additional
+features could be implemented to make the meta-review system more
+sophisticated:
+
+1. A basic ranking list will be built first. In the best case, we should
+   offer an overall ranking list as well as a recent ranking list.
+
+2. Offer an option for users to do meta-reviews directly on gh-board. This
+   might be a bad idea - people shouldn't do meta-review unless they know the
+   whole PR context. This might be a good idea - in the future users can do
```

Comment:
while the 'whole PR context' is important, often a GitHub PR has lots of unnecessary context, like millions of gitmate comments.  gh-board could load only the 'context' needed to do a meta-review, avoiding loading a huge GitHub PR with 100+ comments or just unicorns and needing to flick between diff and discussion.

Reply:
Hmmm, btw do you mean gitmate comments should not be meta-reviewed?

Reply:
Oh, that is a good point.  We dont want gitmate comments to be meta-reviewed, so that is an exception that needs to be built into the system ;-)

But what I mean is that gh-board can have a lighter UI than github for doing meta-reviews.  Typically someone doing a meta-review will not need the 'whole PR context', but only needs a subset.  Which subset they need is going to be hard to guess, but it doesnt matter if they gh-board guess are not always right.  Worst case, the meta-reviewer loads the PR in GitHub and does it there.  gh-board can use 80/20 rule - make the easy ones very easy, and the hard ones it can safely ignore.